.github/workflows/dart.yml

```diff
@@ -56,9 +58,26 @@ jobs:
           release-channel: dev
       - run: dart --version
       - uses: actions/checkout@v2
-      - env:
```

Comment:
I wonder if we could just add support for `--no-pub` to the script

The just "inject" a `pub upgrade` inline here

The trick: if `pub get/upgrade` fails, we DON'T want to run the rest of the steps for that package

Reply:
Ah!

```yaml
- id: pub_mono_repo
  run: working directory...pub upgrade
...
- name: "mono_repo; dart analyzer --fatal-infos ."
  if: ${{ steps.pub_mono_repo.conclusion == 'success' }}
```

or something

Reply:
That was my thought as well, although I don't know how we would manage the part of not running the other jobs. I think maybe we just accept that they will fail with weird errors, it shouldn't be a big issue?

The first step to fail would still be the most informative one.

Reply:
Or possibly we can use some git actions magic to pass this information along

Reply:
Just pushed a change to try this out